internal/provider/resource_bootstrap_git.go

```diff
@@ -0,0 +1,1292 @@
+				CustomType:  customtypes.URLType{},
+				Description: "Url of git repository to bootstrap from.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					validators.URLScheme("http", "https", "ssh"),
+				},
+			},
+			"interval": schema.StringAttribute{
+				CustomType:  customtypes.DurationType{},
+				Description: "Interval at which to reconcile from bootstrap repository.",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					planmodifiers.DefaultStringValue(time.Minute.String()),
+				},
+			},
+			"path": schema.StringAttribute{
+				Description: "Path relative to the repository root, when specified the cluster sync will be scoped to this path.",
+				Optional:    true,
+			},
+			"branch": schema.StringAttribute{
+				Description: "Branch in repository to reconcile from.",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+					planmodifiers.DefaultStringValue(defaultBranch),
+				},
+			},
+			"recurse_submodules": schema.BoolAttribute{
+				Description: "Configures the GitRepository source to initialize and include Git submodules in the artifact it produces.",
+				Optional:    true,
+			},
+			"secret_name": schema.StringAttribute{
+				Description: "Name of the secret the sync credentials can be found in or stored to.",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+					planmodifiers.DefaultStringValue(defaultOpts.Namespace),
+				},
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexp.MustCompile(rfc1123DomainRegex), rfc1123DomainError),
+					stringvalidator.LengthAtMost(253),
+				},
+			},
+			"author_name": schema.StringAttribute{
+				Description: "Author name for Git commits.",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					planmodifiers.DefaultStringValue(defaultAuthor),
+				},
+			},
+			"author_email": schema.StringAttribute{
+				Description: "Author email for Git commits.",
+				Optional:    true,
+			},
+			"gpg_key_ring": schema.StringAttribute{
+				Description: "GPG key ring for signing commits.",
+				Optional:    true,
+			},
+			"gpg_passphrase": schema.StringAttribute{
+				Description: "Passphrase for decrypting GPG private key.",
+				Optional:    true,
+				Sensitive:   true,
+			},
+			"gpg_key_id": schema.StringAttribute{
+				Description: "Key id for selecting a particular key.",
+				Optional:    true,
+			},
+			"commit_message_appendix": schema.StringAttribute{
+				Description: "String to add to the commit messages.",
+				Optional:    true,
+			},
+			"kustomization_override": schema.StringAttribute{
+				Description: "Kustomization to override configuration set by default.",
+				Optional:    true,
```

Comment:
I'm not sure how is this used, will people give the whole YAML including `resources`, if so what happens with patches coming from other files, do we commit any random file?

Reply:
I think this would be fine as it is, if we can check that the gotk- files are in resources, if not, error out to help people. I really like that now people can use all the examples from the docs to customise Flux, awesome job @phillebaba!

Reply:
Sounds like a good idea, as it would cut down on issues. Will implement verification of the override content.